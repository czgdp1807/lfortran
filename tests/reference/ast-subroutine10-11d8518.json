{
    "basename": "ast-subroutine10-11d8518",
    "cmd": "lfortran --indent --show-ast --no-color {infile} -o {outfile}",
    "infile": "tests/subroutine10.f90",
    "infile_hash": "de16e99a96162482992b5d88a4d803f4389d34aa6fc5809a068f8d36",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast-subroutine10-11d8518.stdout",
    "stdout_hash": "52d789989796a69dd4e0b8c4f1f4a2722ba7fcd9e2bd4121af14e494",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}