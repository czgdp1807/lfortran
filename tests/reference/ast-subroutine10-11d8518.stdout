(TranslationUnit
    [(Program
        test
        (TriviaNode
            []
            [(EndOfLine)
            (EndOfLine)]
        )
        []
        []
        [(Declaration
            (AttrType
                TypeReal
                []
                ()
                None
            )
            []
            [(x
            []
            []
            ()
            None
            ())]
            (TriviaNode
                []
                [(EndOfLine)
                (EndOfLine)]
            )
        )]
        [(SubroutineCall
            0
            sub
            []
            [(()
            x
            ()
            0)
            (()
            1
            ()
            0)
            (()
            ()
            ()
            100)
            (()
            ()
            ()
            200)]
            []
            ()
        )
        (Print
            0
            ()
            [(String
                "Success:"
            )
            x]
            ()
        )
        (Stop
            0
            ()
            ()
            (TriviaNode
                []
                [(EndOfLine)
                (EndOfLine)]
            )
        )
        (Print
            100
            ()
            [(String
                "Negative input value"
            )]
            ()
        )
        (Stop
            0
            ()
            ()
            (TriviaNode
                []
                [(EndOfLine)
                (EndOfLine)]
            )
        )
        (Print
            200
            ()
            [(String
                "Input value too large"
            )]
            ()
        )
        (Stop
            0
            ()
            ()
            (TriviaNode
                []
                [(EndOfLine)
                (EndOfLine)]
            )
        )]
        []
    )
    (Subroutine
        sub
        [(x)
        (i)
        (())
        (())]
        []
        ()
        ()
        []
        []
        []
        [(Declaration
            (AttrType
                TypeReal
                []
                ()
                None
            )
            [(AttrIntent
                Out
            )]
            [(x
            []
            []
            ()
            None
            ())]
            ()
        )
        (Declaration
            (AttrType
                TypeInteger
                []
                ()
                None
            )
            [(AttrIntent
                In
            )]
            [(i
            []
            []
            ()
            None
            ())]
            ()
        )]
        [(If
            0
            ()
            (< i 0)
            [(Return
                0
                1
                ()
            )]
            []
            ()
            ()
            ()
        )
        (If
            0
            ()
            (> i 10)
            [(Return
                0
                2
                ()
            )]
            []
            ()
            ()
            ()
        )
        (=
            0
            x
            i
            ()
        )]
        []
    )]
)
